### PR TITLE
update dependencies, add new rules introduced by updates

### PR DESCRIPTION
**Pull request by ndhoule**
This changeset updates eslint plugin dependencies and adds a few new rules that should cause a minimum of non-fixable errors.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
+# vNEXT
+
+## Removed rules:
+
+- [`@typescript-eslint/ban-ts-ignore` (error)](https://github.com/typescript-eslint/typescript-eslint/blob/4670aabef31d9017ad302f206b9c2f18d53c8ee4/packages/eslint-plugin/docs/rules/ban-ts-ignore.md): Replaced by the newer, stricter `ban-ts-comment` rule. If your code overrides this rule, something like `ag -Q '@typescript-eslint/ban-ts-ignore' -l --print0 | xargs -0 sed -i '' 's|@typescript-eslint/ban-ts-ignore|@typescript-eslint/ban-ts-comment|g'` should fix most instances of it.
```
Comment (serhalp) [inline CHANGELOG.md]: :jonathanb:

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+
+## New rules:
```
Comment (serhalp) [inline CHANGELOG.md]: maybe update the PR title to clarify this also adds rules?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+- [`@typescript-eslint/no-extra-semi` (error)](https://github.com/typescript-eslint/typescript-eslint/blob/4670aabef31d9017ad302f206b9c2f18d53c8ee4/packages/eslint-plugin/docs/rules/no-extra-semi.md)
+- [`@typescript-eslint/no-implied-eval` (error)](https://github.com/typescript-eslint/typescript-eslint/blob/4670aabef31d9017ad302f206b9c2f18d53c8ee4/packages/eslint-plugin/docs/rules/no-implied-eval.md)
+- [`@typescript-eslint/no-throw-literal` (error)](https://github.com/typescript-eslint/typescript-eslint/blob/4670aabef31d9017ad302f206b9c2f18d53c8ee4/packages/eslint-plugin/docs/rules/no-throw-literal.md)
+- [`@typescript-eslint/prefer-as-const` (error)](https://github.com/typescript-eslint/typescript-eslint/blob/4670aabef31d9017ad302f206b9c2f18d53c8ee4/packages/eslint-plugin/docs/rules/prefer-as-const.md)
+- [`@typescript-eslint/no-unnecessary-boolean-literal-compare` (error)](https://github.com/typescript-eslint/typescript-eslint/blob/4670aabef31d9017ad302f206b9c2f18d53c8ee4/packages/eslint-plugin/docs/rules/no-unnecessary-boolean-literal-compare.md)
```
Comment (serhalp) [inline CHANGELOG.md]: :partyparrot:

File: CHANGELOG.md (hunk 1 above)
Comment (ndhoule) [reply CHANGELOG.md]: because I `sed` so

**Pull request merged**